tests/test_outputdir.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+  Copyright 2023 SINTEF Digital, Mathematics and Cybernetics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+
+#define BOOST_TEST_MODULE TestOutputDir
+#define BOOST_TEST_NO_MAIN
+#include <boost/test/unit_test.hpp>
+
+#include <opm/simulators/flow/Main.hpp>
+
+#include <filesystem>
+#include <fstream>
+#include <string>
+
+namespace {
+
+struct Fixture {
+    Fixture()
+    {
+        const std::string deck = R"(RUNSPEC
+DIMENS
+  10 10 3 /
+START
+ 8 OCT 2020 /
+GRID
+DXV
+  10*100.0 /
+DYV
+  10*100.0 /
+DZV
+  3*10.0 /
+DEPTHZ
+  121*2000.0 /
+PERMX
+  300*100.0 /
+PERMY
+  300*100.0 /
+PERMZ
+  300*10.0 /
+PORO
+  300*0.3 /
+PROPS
+SOLUTION
+SCHEDULE
+
+TSTEP
+  10
+/
+END
+)";
+
+        input_path = std::filesystem::temp_directory_path() / "outputdir_test/";
+
+        std::filesystem::remove_all(input_path);
+        std::filesystem::create_directory(input_path);
+        std::filesystem::create_directory(input_path / "subdir");
+        std::filesystem::create_directory(input_path / "subdir" / "subdir");
```

Comment:
You could collapse this to
```C++
std::filesystem::create_directories(input_path / "subdir" / "subdir")
```
which is the moral equivalent of `mkdir -p`.

Reply:
Done.